### PR TITLE
Ade dev week4

### PR DESCRIPTION
**Pull request by ade-ademiluyi**

### CODE REVIEW

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/data/Task.java [hunk 1]
```diff
@@ -26,4 +26,347 @@ public Task(long id, String title, long timestamp) {
...
+    * @param userID the unique code used to identify this specific user.
+    * @return a user object with the properties specified in the Builder.
+  */
+  public Game retrieveUser(String userID) {
+    Key userEntityKey = KeyFactory.createKey("User", userID);
+    Entity userEntity;
+    try {
+      userEntity = datastore.get(userEntityKey);
+    }
+    catch(Exception e) {
```
Comment (shenqiuz) [inline walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/data/Task.java:352]: https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html#get-com.google.appengine.api.datastore.Key-

The DatastoreService#get(Key) method will throw an EntityNotFoundException. So you can re-throw the exception like below 

public Game retrieveUser(String userID) throws EntityNotFoundException {
   Key userEntityKey = KeyFactory.createKey("User", userID);
    Entity userEntity = datastore.get(userEntityKey);
    .....
}

In this case, the caller of this method should capture the EntityNotFoundException

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -8,16 +8,98 @@
...
+import java.util.Collections;
+
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+
+        //THIS GETS THE LIST OF OPTIONAL ATTENDEES FOR THE REQUESTED MEETING
```
Comment (shenqiuz) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:27]: Don't use all-capital letters for comments.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -8,16 +8,98 @@
...
+        Set<String> optionalAttendees = new HashSet<>();
+        for(String attendee: request.getOptionalAttendees()) {
+            optionalAttendees.add(attendee);
+        }
+        
+        //THIS GETS THE LIST OF ATTENDEES FOR THE REQUESTED MEETING
```
Comment (shenqiuz) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:33]: Don't use all-capital letters for comments.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -8,16 +8,98 @@
...
+        if (mandatoryAttendees.size() == 0) 
+        {
+            ArrayList<TimeRange> timesExcluded = new ArrayList<>();
+            //CHECKS IF the optional attendees have any other event then stotimeBetween that time to avoid
+            for(Event event: events) {
+                if(!Collections.disjoint(event.getAttendees(), optionalAttendees)) {
+                    timesExcluded.add(event.getWhen());
+                }
+            }
+            //
```
Comment (shenqiuz) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:52]: Delete empty comment.

File: portfolio/src/main/java/com/google/sps/servlets/deleteDataServlet.java [hunk 5]
```diff
@@ -16,23 +16,41 @@
...
+import com.google.sps.data.Comment;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet responsible for deleting tasks. */
```
[the remark this answers is not in the document]
Comment (ade-ademiluyi) [reply portfolio/src/main/java/com/google/sps/servlets/deleteDataServlet.java:27]: sorry didn't catch that

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/data/Task.java (hunk 1 above)
Comment (ade-ademiluyi) [reply walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/data/Task.java:352]: sorry this piece of code wasn't meant to be included

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/data/Task.java [hunk 6]
```diff
@@ -26,4 +26,347 @@ public Task(long id, String title, long timestamp) {
...
+    * @param hintID the unique code used to identify this specific hint.
+    * @return a Hint object with the properties specified in the Builder.
+  */
+  public Hint retrieveHint(String hintID) {
+    Key hintEntityKey = KeyFactory.createKey("Hint", HintID);
+    Entity hintEntity;
+    try {
+      hintEntity = datastore.get(hintEntityKey);
+    }
+    catch(Exception e) {
```
[the remark this answers is not in the document]
Comment (ade-ademiluyi) [reply walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/data/Task.java:281]: sorry this piece of code wasn't meant to be included

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/data/Task.java [hunk 7]
```diff
@@ -26,4 +26,347 @@ public Task(long id, String title, long timestamp) {
...
+    * Updates an entity of a single hint data in datastore
+    * @param hint a Hint variable representing a single instance of a hint.
+  */
+  public void updateHint(Hint hint) {
+    Key hintEntityKey = KeyFactory.createKey("Hint", hint.getHintID());
+    Entity hintEntity;
+    try {
+      hintEntity = datastore.get(hintEntityKey);
+    }
+    catch(Exception e) {
```
[the remark this answers is not in the document]
Comment (ade-ademiluyi) [reply walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/data/Task.java:260]: sorry this piece of code wasn't meant to be included

**Pull request closed without merging**